### ISSUE TITLE
Still missing LICENSE 

### ISSUE THREAD

**Issue by freepdk** [free-pdk/free-pdk-examples#3]
Nobody can freely use the samples and / or contribute to this project when there is no clear LICENSE.

**Comment by serisman** [comment]
I am still trying to decide what license to apply to these examples.

My preference would be MIT (or maybe even public domain), but I need confirmation from at least @freepdk if that would be ok.

Most of these are based on the Arduino built-in examples which are in the public domain, but there is at least some code (primarily serial.h / Serial_HelloWorld) that has roots from the easy-pdk-programmer-software repo which is GPL v3.

The include/easy-pdk and include/pdk directories are copies of the easy-pdk-includes and pdk-includes repos, which will have their own licenses.  Not sure how to license those yet either.  My preference for at least the pdk-includes is still MIT, but I'm not as sure about easy-pdk-includes.

So, @freepdk, are you ok with these examples being licensed as MIT, or would you require keeping at least the one licensed under GPL v3?

**Comment by freepdk** [comment]
Why PD or MIT?
You want others to be able to use this work and build it into their closed source products?
[ ] yes
[ ] no

The project is a TOY project. Everybody should have fun with it and learn. 
GPL assures exactly this. Whatever is created out of this must be shared. This means more people can participate from it.

For commercial use I recommend to use the vendor supplied tools.

**Comment by serisman** [comment]
> Why PD or MIT?
> You want others to be able to use this work and build it into their closed source products?
> [ ] yes
> [ ] no

Sure, why not?

That is still increased exposure and can still lead to more potential users hunting down bugs and contributing enhancements.

> The project is a TOY project. Everybody should have fun with it and learn.
> GPL assures exactly this. Whatever is created out of this must be shared. This means more people can participate from it.

As far as I understand it (one such source: https://lukasa.co.uk/2012/05/GPL_vs_MIT_Which_License_To_Use/), the GPL is more restrictive in that it requires any use to also be shared back and kept under a GPL license.  This sounds like it would encourage more sharing, but it may also deter usage to begin with.

MIT is more permissive and really only stipulates that there is attribution to the original author(s).

There are a lot of companies with closed source products, that still contribute back to the open source community.

> For commercial use I recommend to use the vendor supplied tools.

Sure, in its current state that may be wise, but why restrict it?

I can appreciate that you have put a LOT of your own time into the programmer (and supporting documentation, etc...), and don't want someone to just come in and sell everything you have come up with.  That certainly makes sense for something like the programmer, but does that really matter for some example code (and include files)?

**Comment by freepdk** [comment]
> I can appreciate that you have put a LOT of your own time into the programmer (and supporting documentation, etc...), and don't want someone to just come in and sell everything you have come up with. That certainly makes sense for something like the programmer, but does that really matter for some example code (and include files)?

You are completely wrong here. I have nothing against anything from the project is sold. 
GPL also does not forbid this. 
The only thing required from GPL, is to include the full source of the original and all modifications.
What other "more permissive" you need / want to grant?
Why remove this simple restriction by choosing PD or MIT?

> > Why PD or MIT?
> > You want others to be able to use this work and build it into their closed source products?
> > [ ] yes
> > [ ] no
> 
> Sure, why not?
> 
> That is still increased exposure and can still lead to more potential users hunting down bugs and contributing enhancements.
I see it exact the other way. By choosing MIT or PD you might discourage several free opens source developers to contribute new features or bug fixes since there is no way fro them to protect their work from being included in close source products.

Almost 20 years ago, I had almost the same thinking and released anything under MIT. There was also no argument back then which could have convinced me to use GPL. I learned it the hard way that I was wrong. 

So for me: sharing = caring, only GPL really ensures sharing.

**Comment by serisman** [comment]
Well, I am not particularly interested in (or qualified for) an extended discussion on this.

If you feel strongly that GPL is the way to go, I'm fine with that, and can keep my preferences to myself.

I don't see a big difference on a repo like this one anyway.

It might be more important on the pdk-includes repo, because that is more likely to be needed by and included into other projects.  I don't like forcing people to license their projects as GPL (or even sharing them back) just because they use the common device include files.  But, as Philipp mentioned on one of the other repos, we may have to stick with GPL2+LE anyway in order to be considered for inclusion in SDCC.  We can address this on that repo.

**Comment by freepdk** [comment]
> It might be more important on the pdk-includes repo, because that is more likely to be needed by and included into other projects. I don't like forcing people to license their projects as GPL (or even sharing them back) just because they use the common device include files. But, as Philipp mentioned on one of the other repos, we may have to stick with GPL2+LE anyway in order to be considered for inclusion in SDCC. We can address this on that repo.

I fully agree to use the SDCC compatible license (GPL2+LE) for all of the header files which are supposed to be merged to SDCC.

**Comment by 1500WK1500** [comment]
I am planning to add platformio support for the padauk microcontrollers (with the free-pdk toolchain).
My platform project is currently licensed under Apache 2.0.
[https://github.com/1500WK1500/platform-padauk](https://github.com/1500WK1500/platform-padauk)

I would like to add your examples to the project. However there is still no licence file in this repository.
Am I allowed to add the examples into my project?
